types/src/byzantine.rs

```diff
@@ -70,53 +63,61 @@ impl FraudProof for BadEncodingFraudProof {
             );
         }
 
-        let merkle_row_roots = header.dah.row_roots();
-        let merkle_col_roots = header.dah.column_roots();
-
         // NOTE: shouldn't ever happen as header should be validated before
-        if merkle_row_roots.len() != merkle_col_roots.len() {
+        if header.dah.row_roots().len() != header.dah.column_roots().len() {
             bail_validation!(
                 "dah rows len ({}) != dah columns len ({})",
-                merkle_row_roots.len(),
-                merkle_col_roots.len(),
+                header.dah.row_roots().len(),
+                header.dah.column_roots().len(),
             );
         }
 
-        if usize::from(self.index) >= merkle_row_roots.len() {
+        let square_width = usize::from(header.dah.square_width());
+        let ods_width = square_width / 2;
+
+        if usize::from(self.index) >= square_width {
             bail_validation!(
-                "fraud proof index ({}) >= dah rows len ({})",
+                "fraud proof index ({}) >= dah square width ({})",
                 self.index,
-                merkle_row_roots.len()
+                square_width,
             );
         }
 
-        if self.shares.len() != merkle_row_roots.len() {
+        if self.shares.len() != square_width {
             bail_validation!(
-                "fraud proof shares len ({}) != dah rows len ({})",
+                "fraud proof shares len ({}) != dah square width ({})",
                 self.shares.len(),
-                merkle_row_roots.len()
+                square_width,
             );
         }
 
-        let root = match self.axis {
-            AxisType::Row => merkle_row_roots[usize::from(self.index)].clone(),
-            AxisType::Col => merkle_col_roots[usize::from(self.index)].clone(),
-        };
-
-        // verify if the root can be converted to a cid and back
-        let mh = Multihash::wrap(NMT_MULTIHASH_CODE, &root.to_array())?;
-        let cid = Cid::new_v1(NMT_CODEC, mh);
-        let root = NamespacedHash::try_from(cid.hash().digest())?;
+        let shares_count = self.shares.iter().filter(|s| s.is_some()).count();
+        if shares_count < ods_width {
+            bail_validation!(
+                "fraud proof non-nil shares count ({shares_count}) < dah ods width ({ods_width})"
```

Comment:
```suggestion
                "fraud proof present shares count ({shares_count}) < dah ods width ({ods_width})"
```
I think this error should be translated to rust, imagine seeing some error in Rust saying like 'value is non-nil' :p